Riot/Modules/Room/MXKRoomViewController.m

```diff
@@ -2517,6 +2518,126 @@ - (void)updateCurrentEventIdAtTableBottom:(BOOL)acknowledge
     }
 }
 
+- (BOOL)eligibleForReadMarkerUpdate:(MXEvent *)event {
+    // Prevent the readmarker to be placed on a relatesTo or a redaction event
+    if (event.relatesTo || event.redacts)
+    {
+        return NO;
+    }
+    
+    return YES;
+}
+
+- (BOOL)isEventPosteriorToCurrentReadMarker:(MXEvent *)event {
+    if (roomDataSource.room.accountData.readMarkerEventId)
+    {
+        MXEvent *currentReadMarkerEvent = [roomDataSource.mxSession.store eventWithEventId:roomDataSource.room.accountData.readMarkerEventId inRoom:roomDataSource.roomId];
+        if (!currentReadMarkerEvent)
+        {
+            currentReadMarkerEvent = [roomDataSource eventWithEventId:roomDataSource.room.accountData.readMarkerEventId];
+        }
+        
+        // Update the read marker only if the current event is available, and the new event is posterior to it.
+        return currentReadMarkerEvent && (currentReadMarkerEvent.originServerTs <= event.originServerTs);
+    }
+    return YES;
+}
+
+/// Try to update the read marker by looking for an eligible event displayed at the bottom of the tableview
+- (void)updateReadMarkerEvent
+{
+    // Compute the content offset corresponding to the line displayed at the table bottom (just above the toolbar).
+    CGFloat contentBottomOffsetY = _bubblesTableView.contentOffset.y + (_bubblesTableView.frame.size.height - _bubblesTableView.adjustedContentInset.bottom);
```

Comment:
I’m not sure. I used the same mechanism as the one used to find the event to acknowledge because the read marker was updated in the same process as the acknowledgement.